### PR TITLE
Feature/signup page

### PR DESCRIPTION
**Pull request by kanishkalakmina**

### CODE REVIEW

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/auth/Auth.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  return (
+    <div className={styles['container']}>
+      <div className={styles['main-login']}>
+        <div className={styles['login-contain']}>
+          <div className={styles['left-side']}>
+            {isRegister ? <SignUp /> : <Login />}
+            <div className={styles['toggle-text']}>
+              {isRegister ? (
+                <span style={{ color: '#000' }}>
+                  {'Already have an account? '}{' '}
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/auth/Auth.tsx:22]: remove {' '} part

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/auth/Auth.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                    }}
+                    onClick={toggleRegister}
+                  >
+                    Login here
+                  </span>
+                </span>
+              ) : (
+                <span style={{ color: '#000' }}>
+                  {"Don't have an account? "}
+                  {'  '}
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/auth/Auth.tsx:37]: remove {' '}

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/auth/Auth.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    setIsRegister(!isRegister);
+  };
+  return (
+    <div className={styles['container']}>
+      <div className={styles['main-login']}>
+        <div className={styles['login-contain']}>
+          <div className={styles['left-side']}>
+            {isRegister ? <SignUp /> : <Login />}
+            <div className={styles['toggle-text']}>
+              {isRegister ? (
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/auth/Auth.tsx:20]: don't need to render a whole block here. The only difference of two blocks are the labels. Just render labels conditionally.
{isRegister ? Login here : Sign up now}

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/auth/Auth.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                  >
+                    Sign up now
+                  </span>
+                </span>
+              )}
+            </div>
+          </div>
+          <div className={styles['right-side']}>
+            <div className={styles['welcomeNote']}>
+              {/* <h3>Welcome to Bistec Global</h3> */}
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/auth/Auth.tsx:54]: remove comment

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/login/state.ts [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import { useRouter } from 'next/router';
+// eslint-disable-next-line @nrwl/nx/enforce-module-boundaries
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/login/state.ts:2]: why this line?

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/signup/state.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import { useRouter } from 'next/router';
+// eslint-disable-next-line @nrwl/nx/enforce-module-boundaries
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/signup/state.ts:2]: why this?

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/signup/state.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import type {
+  UserRequest,
+  UserResponse,
+} from 'apps/blockchain-frontend/interfaces/viewModels';
+import { CreateUserByEmail } from '../../api/fetchData';
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/signup/state.ts:7]: use standard file path

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/signup/state.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+export function useComponentState() {
+  const router = useRouter();
+
+  const handleSignupFormSubmit = async (e) => {
+    e.preventDefault();
+    const raw: UserRequest = {
+      Email: e.target.email.value,
+      Password: e.target.password.value,
+    };
+    validateUser(raw);
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/signup/state.ts:18]: rename this to `createUser`

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/signup/state.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  async function validateUser(formData: UserRequest): Promise<void> {
+    const user: UserResponse = await CreateUserByEmail(formData);
+    if (!user) {
+      console.error('User not found');
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/signup/state.ts:24]: console.error('User not created!')

**Review by gayashan4lk — CHANGES_REQUESTED** [review]

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/login/state.ts (hunk 5 above)
Comment (kanishkalakmina) [reply src/portals/blockchain-frontend/apps/blockchain-frontend/components/login/state.ts:2]: import type {
  UserRequest,
  UserResponse,
} from 'apps/blockchain-frontend/interfaces/viewModels';  

getting error in this import 

Projects cannot be imported by a relative or absolute path, and must begin with a npm scopeeslint[@nrwl/nx/enforce-module-boundaries](file:///)

**Pull request merged**